public/app/features/explore/state/epics/setPausedStateEpic.ts

```diff
@@ -0,0 +1,21 @@
+import { Epic } from 'redux-observable';
+import { ActionOf } from 'app/core/redux/actionCreatorFactory';
+import { flatMap } from 'rxjs/operators';
+
+import { setPausedStateAction, SetPausedStatePayload } from '../actionTypes';
+import { runQueriesAction } from '../actionTypes';
+import { EMPTY, of } from 'rxjs';
+
+export const setPausedStateEpic: Epic<ActionOf<any>> = action$ => {
```

Comment:
Not sure if it makes sense this way as the epics and actions are bit mixed. Ideally I would map this to something like `startQueries` and `pauseQueries` actions and then also have epic that would convert all other actions that are pausing queries to `pauseQueries` which would then be piped into `runQueriesBatchEpic` as a pause signal.